File: scratch/read_users.mjs
import { google } from 'googleapis';
import fs from 'fs';

const creds = JSON.parse(fs.readFileSync('./google-credentials.json', 'utf8'));

const auth = new google.auth.JWT(
  creds.client_email,
  null,
  creds.private_key,
  ['https://www.googleapis.com/auth/spreadsheets']
);

const sheets = google.sheets({ version: 'v4', auth });
const MASTER_SS_ID = "1IQCs35RQlMMQsGB-CRczJeuRqa8WIxW4Sy_kjZyHP2M";

async function run() {
  try {
    const res = await sheets.spreadsheets.values.get({
      spreadsheetId: MASTER_SS_ID,
      range: 'Users!A:F'
    });
    console.log("Users Rows:", res.data.values);
  } catch (err) {
    console.error("Failed:", err.message);
  }
}

run();

File: scratch/read_users.py
import os
import sys
import gspread
from google.oauth2.service_account import Credentials

def main():
    creds_path = os.path.expanduser("~/.gemini/configs/keys/ga-projects-key.json")
    if not os.path.exists(creds_path):
        print(f"Error: Creds file not found at {creds_path}")
        sys.exit(1)
        
    creds = Credentials.from_service_account_file(creds_path, scopes=["https://www.googleapis.com/auth/spreadsheets"])
    client = gspread.authorize(creds)
    
    sheet_id = "1IQCs35RQlMMQsGB-CRczJeuRqa8WIxW4Sy_kjZyHP2M"
    try:
        sh = client.open_by_key(sheet_id)
        worksheet = sh.worksheet("Users")
        values = worksheet.get_all_values()
        for idx, row in enumerate(values):
            print(f"[{idx}] {row}")
    except Exception as e:
        print(f"Exception: {e}")

if __name__ == "__main__":
    main()
